beacon-chain/db/kv/wss.go

```diff
@@ -0,0 +1,89 @@
+package kv
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	types "github.com/prysmaticlabs/eth2-types"
+	statev2 "github.com/prysmaticlabs/prysm/beacon-chain/state/v2"
+	"github.com/prysmaticlabs/prysm/config/params"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1/wrapper"
+)
+
+// SaveOrigin loads an ssz serialized Block & BeaconState from an io.Reader
+// (ex: an open file) prepares the database so that the beacon node can begin
+// syncing, using the provided values as their point of origin. This is an alternative
+// to syncing from genesis, and should only be run on an empty database.
+func (s *Store) SaveOrigin(ctx context.Context, stateReader, blockReader io.Reader) error {
```

Comment:
I guess one thing that is tricky and that we need to be aware of would be what if a user provides a phase0/altair/merge WSS state ? Currently the code assumes that the user only provides an altair block/state.

Reply:
Agreed, copy/pasting from my response on this to Raul: 
Working on making this more future proof in cp-sync-sniff-origin-init 65912a9
Still need to do a bit of testing but i think that work will decouple this code from relying on a particular proto struct.